install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -3150,10 +3150,24 @@ authentication:
         enabled: true
         # -- SPIRE namespace to install into
         namespace: cilium-spire
+        # -- init container image of SPIRE agent and server
+        initImage:
+          override: ~
+          repository: "docker.io/library/busybox"
+          tag: "1.35.0"
+          digest: "sha256:223ae047b1065bd069aac01ae3ac8088b3ca4a527827e283b85112f29385fb1b"
+          useDigest: true
+          pullPolicy: "Always"
         # SPIRE agent configuration
         agent:
           # -- SPIRE agent image
-          image: ghcr.io/spiffe/spire-agent:1.6.3@sha256:8eef9857bf223181ecef10d9bbcd2f7838f3689e9bd2445bede35066a732e823
+          image:
```

Comment:
Changing the type and re-using the name of the existing values is a breaking change as we do not give users a deprecation period.

Deprecating the old value and introducing a new one (with a different name) is what we normally do. But i agree that we should use the property `image` to be consistent with other definitions of images.

@kaworu what's the recommended approach in such cases from a sig-helm perspective? Would the breaking change be ok for v1.15 (without backporting it to v1.14)?

Reply:
indeed, I also hesitated to change a new name inconsistent with other definitions of images

Reply:
> what's the recommended approach in such cases from a sig-helm perspective?

Generally speaking, when possible: go through a deprecation stage introducing the alternative for the next version n and document in upgrade.rst, then remove the deprecated option in n+1.

Here it's tricky because we want both the new and old values under the same `image:` key, but we could check in Helm the `typeOf` `image` and use the "old" or "new" behaviour when it's a string or an object, respectively.

> Would the breaking change be ok for v1.15 (without backporting it to v1.14)?

Since this block is about Mutual Authentication configuration, and given [Mutual Authentication is in Beta maturity stage](https://docs.cilium.io/en/latest/network/servicemesh/mutual-authentication/mutual-authentication/), I think it's fine to break it for v1.15.

Reply:
@kaworu Nice - i wasn't aware of the `typeOf` function in Helm. Thanks!

@weizhoublue Given this knowledge, I'd recommend to introduce this change backwards-compatible.

Something like this should work.
```
{{- if eq (typeOf .Values.authentication.mutual.spire.install.agent.image) "string"}}
image: {{ .Values.authentication.mutual.spire.install.agent.image }}
{{- else }}
image: {{ include "cilium.image" .Values.authentication.mutual.spire.install.agent.image | quote }}
imagePullPolicy: {{ .Values.authentication.mutual.spire.install.agent.image.pullPolicy }}
{{- end }}
```

At least a quick local test seems to work. The only thing that irritates me a little bit is the output when "overriding" the value with a different type 

```
coalesce.go:220: warning: cannot overwrite table with non table for cilium.authentication.mutual.spire.install.agent.image
```

:thinking:

Reply:
OK, it would be a graceful manner for backwards-compatible